homework1/Program.cs

```diff
@@ -0,0 +1,133 @@
+﻿if (!TestTransformation() || !TestReverseConversion() || TestIsARightSymbol())
+{
+      return -1;
+}
+
+System.Console.Write("Enter a string consisting only of letters or numbers: ");
+string? line = System.Console.ReadLine();
+if (line is null)
+{
+      return -1;
+}
+
+if (!IsARightSymbol(line))
+{
+      return -1;
+}
+
+var (result, index) = Transformation(line);
+System.Console.WriteLine("Result: " + result + " index: " + index);
+string originLine = ReverseConversion(result, index);
+System.Console.WriteLine(originLine);
+System.Console.WriteLine(line == originLine);
+
+bool IsARightSymbol(string str)
+{
+      for (int i = 0; i < str.Length; ++i)
+      {
+            if (!char.IsLetterOrDigit(str[i]))
+            {
+                  return false;
+            }
+      }
+
+      return true;
+}
+
+(string TmpString, int Index) Transformation(string line)
+{
+      int length = line.Length;
+      var rotations = new List<string>(length);
+      for (int i = 0; i < length; ++i)
+      {
+            string rotation = line.Substring(i) + line.Substring(0, i);
+            rotations.Add(rotation);
+      }
+
+      var sortedRotations = rotations.OrderBy(r => r).ToList();
+      foreach (string i in sortedRotations)
+      {
+            System.Console.WriteLine(i);
+      }
+
+      char[] tmpString = new char[length];
```

Comment:
```suggestion
      var tmpString = new char[length];
```
Не пишите один и тот же тип в объявлении дважды